lib/puppet/application.rb

```diff
@@ -357,10 +357,6 @@ def setup
   end
 
   def setup_logs
-    if options[:debug] || options[:verbose]
-      Puppet::Util::Log.newdestination(:console)
-    end
-
```

Comment:
This modifies the API for puppet application, for all modules that derive from it and have expectations on how it behaves.
Not all such applications are in the puppet code base.